WalletWasabi.Fluent/ViewModels/Wallets/Send/TransactionPreviewViewModel.cs

```diff
@@ -436,7 +438,10 @@ private async Task OnConfirmAsync()
 
 			_info.UserLabels = result.Result;
 
-			var transaction = await Task.Run(() => TransactionHelpers.BuildTransaction(_wallet, _info));
+			var transaction = _info.IsOptimized ?
+				await Task.Run(() => TransactionHelpers.BuildChangelessTransaction(_wallet, _info.Address, _info.UserLabels, _info.FeeRate, _info.ChangelessCoins))
```

Comment:
Why do we need `_info.ChangelessCoins` variable? Without more context, I would guess that one can just use `_info.Coins`. 

Do we need `_info.Coins` and `_info.ChangelessCoins` to compare fees for the respective txs ?

Reply:
If my understanding is correct, `_info.Coins` and `_info.ChangelessCoins` are the same, but the `_info.IsOptimized` will only be `true` if `_info.ChangelessCoins` is not empty.

Reply:
> Why do we need `_info.ChangelessCoins` variable? Without more context, I would guess that one can just use `_info.Coins`.

Unfortunately it doesn't use the `_info.Coins`.
The reason we "need" to use `_info.ChangelessCoins` is that
`public bool IsOptimized => ChangelessCoins.Any();`.

So if we want to use `IsOptimized`, we need save the selected coins to `ChanglessCoins` prop.

Reply:
Well, but you can assign `_info.IsOptimized` to `true` and set `_info.Coins` and you don't need `_info.ChangelessCoins`...? It seems even conceptually simpler to me.

Reply:
Yep, I totally get your point. I'm not sure what was the original logic behind those props, but we don't use them anywhere.
Anyway I'm gonna change them.